### PR TITLE
server: remove etcd

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: ref #136

Problem Summary: Removed etcd. ETCD's HTTP api may conflict with naming of serverless tier

But etcd flags and `/api` prefix are left. `/api` prefix will be removed in the next PR for new Handler api.

What is changed and how it works:

1. a btree is used to replace etcd for persistence(in memory only)
2. a refactor on code of config manager to reduce the complexity

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
I hooked a line in `LoggerManager.watchCfg`
```
[xhe@PC server]$ echo '{"level":"info","log-file":{"max-size":300,"max-days":1,"max-backups":1}}' | ../../bin/tiproxyctl config log set
2022/11/29 19:13:20.145 +08:00  INFO    main.lgmgr      logger/manager.go:68    updated cfg     {"cfg": {"level":"info","log-file":{"max-size":300,"max-days":1,"max-backups":1}}}
```
- [ ] No code

Notable changes

- [ ] Has configuration change
- [ ] Has HTTP API interfaces change (Don't forget to [add the declarative for API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- [ ] Has tiproxyctl change
- [ ] Other user behavior changes

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by djshow832** [comment]
Please fix CI.

File: pkg/manager/config/config.go [hunk 1]
```diff
@@ -17,170 +17,75 @@ package config
...
-
-// GetCfgWatch returns the channel that contains updated configuration.
-func GetCfgWatch[T OnlineCfgTypes](e *ConfigManager) chan *T {
-	mt := e.metas[getMetaKey[T]()].(*meta[T])
-	return mt.ch
-}
 
-func (e *ConfigManager) GetProxyConfigWatch() <-chan *config.ProxyServerOnline {
-	return GetCfgWatch[config.ProxyServerOnline](e)
+	return json.Unmarshal(c.Value, val)
```
Comment (djshow832) [inline pkg/manager/config/config.go:64]: val will be a map. Is that expected?

**Review by djshow832 — COMMENTED** [review]
Rest LGTM

File: pkg/manager/config/config.go (hunk 1 above)
Comment (xhebox) [reply pkg/manager/config/config.go:64]: Isn't val passed by the caller? He can pass `&ProxyServerOnline` or maybe a `map[string]any`.

**Review by djshow832 — APPROVED** [review]

**Pull request merged**